go/vt/workflow/sleep_workflow.go

```diff
@@ -69,7 +69,8 @@ func (sw *SleepWorkflow) Run(ctx context.Context, manager *Manager, wi *topo.Wor
 	sw.mu.Lock()
 	sw.manager = manager
 	sw.wi = wi
-	sw.node = NewToplevelNode(wi, sw)
+	sw.node = NewNode()
+	sw.node.AttachToWorkflow(wi, sw)
```

Comment:
One of the reasons would be in the comment in node_test that I'm removing in this PR: "Can't really use NewToplevelNode because it takes a WorkflowInfo." I.e. if we merge AttachToWorkflow() with AddRootNode() we'll force every caller of AddRootNode() to have WorkflowInfo at hand while not everyone have it. And theoretically that can be extended: with the current code as it is anyone can create a hierarchical set of Node objects and display them in the workflow UI. And as long as there are no actions, these nodes won't have to be attached to any workflow (maybe Stop button won't work properly, but the rest will just work).
So overall it doesn't look like merging the functions is good here.